src/Hal/Serializer/ItemNormalizer.php

```diff
@@ -169,7 +170,11 @@ private function getComponents($object, string $format = null, array $context)
     private function populateRelation(array $data, $object, string $format = null, array $context, array $components, string $type): array
     {
         $class = \get_class($object);
-        $attributesMetadata = $this->classMetadataFactory ? $this->classMetadataFactory->getMetadataFor($object)->getAttributesMetadata() : null;
+        $cacheKey = $class.':attribute-metadata';
+
+        $attributesMetadata = \array_key_exists($cacheKey, $this->attributesMetadataCache) ?
+            $this->attributesMetadataCache[$cacheKey] :
+            $this->attributesMetadataCache[$cacheKey] = $this->classMetadataFactory ? $this->classMetadataFactory->getMetadataFor($object)->getAttributesMetadata() : null;
 
```

Comment:
Why not `$attributesMetadata = $this->attributesMetadataCache[$cacheKey] ?? this->classMetadataFactory ? $this->classMetadataFactory->getMetadataFor($object)->getAttributesMetadata() : null;`

Reply:
Yep will do